### PR TITLE
DomainParticipant set/get_default_publisher_qos

### PR DESCRIPTION
**Pull request by laura-eprosima**
* Implementation if set_default_publisher_qos and get_default_publisher_qos
* Add new PIM unittest
* Implementation of PSM default_publisher_qos methods
* Add new PSM unittest

NOTE: This PR must be merged after #1073

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1639/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1405/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1635/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1676/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1438/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1668/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1691/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1452/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1683/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1833/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1580/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1827/)

File: src/cpp/dds/domain/DomainParticipant.cpp [hunk 1]
```diff
@@ -129,15 +129,7 @@ dds::pub::qos::PublisherQos DomainParticipant::default_publisher_qos() const
...
-    ReturnCode_t code = this->delegate()->set_default_publisher_qos(qos);
-    if (code == ReturnCode_t::RETCODE_INCONSISTENT_POLICY)
-    {
-        throw dds::core::InconsistentPolicyError("Inconsistent Qos");
-    }
-    else if (code == ReturnCode_t::RETCODE_IMMUTABLE_POLICY)
-    {
-        throw dds::core::ImmutablePolicyError("Immutable Qos");
-    }
+    this->delegate()->set_default_publisher_qos(qos);
```
Comment (IkerLuengo) [inline src/cpp/dds/domain/DomainParticipant.cpp]: It was OK to check for INCONSISTENT. Also check for UNSUPPORTED.
No need for IMMUTABLE, default_publisher_qos can always be changed

File: src/cpp/dds/pub/Publisher.cpp [hunk 2]
```diff
@@ -63,11 +63,7 @@ void Publisher::qos(
         const qos::PublisherQos& pqos)
 {
     ReturnCode_t code = this->delegate()->set_qos(pqos);
-    if (code == ReturnCode_t::RETCODE_INCONSISTENT_POLICY)
-    {
-        throw dds::core::InconsistentPolicyError("Inconsistent Qos");
-    }
-    else if (code == ReturnCode_t::RETCODE_IMMUTABLE_POLICY)
+    if (code == ReturnCode_t::RETCODE_IMMUTABLE_POLICY)
```
Comment (IkerLuengo) [inline src/cpp/dds/pub/Publisher.cpp:66]: Should check for INCONSISTENT and UNSUPPORTED too, even if it is only to prepare for future changes.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1869/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1610/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1862/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1874/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1615/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1867/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1898/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1636/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1891/)

File: include/fastdds/dds/core/policy/ParameterTypes.hpp [hunk 3]
```diff
@@ -165,22 +166,83 @@ class Parameter_t
     }
 
     /**
-     * Virtual method used to add the parameter to a CDRMessage_t message.
+     * Method used to add the parameter information to a CDRMessage_t message.
      * @param[in,out] msg Pointer to the message where the parameter should be added.
```
Comment (IkerLuengo) [inline include/fastdds/dds/core/policy/ParameterTypes.hpp:170]: Doxydoc for parameter `parameter` missing
Name of second parameter has is `cdr_message`, not `msg`

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1905/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1643/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1898/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1909/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1647/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1902/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1914/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1652/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1907/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1915/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1653/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1908/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1916/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1654/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1909/)

**Comment by richiware** [comment]
Build status:
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1910/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1922/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1660/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1916/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1927/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1665/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1921/)

**Pull request merged**